R/translate.R

```diff
@@ -60,15 +60,33 @@ deepl_translate <- function(path,
     formality,
     values = c("default", "more", "less", "prefer_more", "prefer_less")
   )
-
-  # protect Hugo shortcodes ----
   temp_markdown_file <- withr::local_tempfile()
   markdown_lines <- brio::read_lines(path)
+  temp_markdown_file <- withr::local_tempfile()
+
+  # protect $$ equations $$
+  dollars <- which(grepl("\\$\\$", markdown_lines))
+  dollars_present <- length(dollars > 0)
+  if (dollars_present) {
+    dollars <- matrix(dollars, ncol = 2, byrow = TRUE)
+    undigested_equations <- character(0)
+    index <- 1
+    for(i in 1:nrow(dollars))
+      for(j in (dollars[i,1]+1):(dollars[i,2]-1)) {
+        undigested_equations[index] <- markdown_lines[j]
+        index <- index+1
+        markdown_lines[j] <- sprintf("`%s`", digest::digest(markdown_lines[j]))
```

Comment:
- could this use purrr like the code for shortcodes?
- what happens if a document has both shortcodes and equations?